fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/expression/StringStatementRenderer.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query.expression;
+
+import static com.ibm.fhir.database.utils.query.SqlConstants.FROM;
+import static com.ibm.fhir.database.utils.query.SqlConstants.SELECT;
+import static com.ibm.fhir.database.utils.query.SqlConstants.SPACE;
+import static com.ibm.fhir.database.utils.query.SqlConstants.WHERE;
+
+import java.util.List;
+
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.query.FromClause;
+import com.ibm.fhir.database.utils.query.FromItem;
+import com.ibm.fhir.database.utils.query.GroupByClause;
+import com.ibm.fhir.database.utils.query.HavingClause;
+import com.ibm.fhir.database.utils.query.OrderByClause;
+import com.ibm.fhir.database.utils.query.PaginationClause;
+import com.ibm.fhir.database.utils.query.SelectList;
+import com.ibm.fhir.database.utils.query.WhereClause;
+import com.ibm.fhir.database.utils.query.node.BindMarkerNode;
+import com.ibm.fhir.database.utils.query.node.ExpNode;
+
+/**
+ * Helps to render a select statement as a string
+ */
+public class StringStatementRenderer implements StatementRenderer<String> {
+
+    private static final String NEWLINE = System.lineSeparator();
+
+    // fixed for now...perhaps a variable in the future
+    private final char escapeChar = '+';
+
+    // translator to handle SQL differences among databases
+    private final IDatabaseTranslator translator;
+
+    // pretty-print the statement for easier debug
+    private final boolean pretty;
+
+    // For collecting parameter markers seen when rendering the statement
+    private final List<BindMarkerNode> collectBindMarkersInto;
+
+    /**
+     * Public constructor
+     * @param translator
+     */
+    public StringStatementRenderer(IDatabaseTranslator translator, List<BindMarkerNode> collectBindMarkersInto, boolean pretty) {
+        this.translator = translator;
+        this.collectBindMarkersInto = collectBindMarkersInto;
+        this.pretty = pretty;
+    }
+
+    @Override
+    public String and(String left, String right) {
+        StringBuilder result = new StringBuilder();
+//        result.append(left);
+//        result.append(" AND ");
+//        result.append(right);
+        return result.toString();
+    }
```

Comment:
what's actually happening in this ? I think I've used this successfully where it's output correctly.  The commented code segments don't make a lot of sense. (here and below)

Reply:
this is dead code after a refactor. Will clean up.